feat(superuser): Changed expiry time of SU session

maxiuyuan opened the pull request:
Based on [this discussion](https://github.com/getsentry/sentry/pull/31980#discussion_r813419401) Changed expiry time of superuser session to be 15 mins only during privileged org accesses to 15 mins. Will remain the same otherwise

maxiuyuan committed:
changed session time based on priveledge org access

maxiuyuan committed:
check if get_orgs() exists on user for failling test cases

maxiuyuan committed:
Merge branch 'log_su_access' into su_expiry

maxiuyuan committed:
Merge branch 'log_su_access' into su_expiry

maxiuyuan committed:
Merge branch 'log_su_access' into su_expiry

JoshFerge commented on src/sentry/auth/superuser.py:
```diff
@@ -82,6 +96,11 @@ def __init__(self, request, allowed_ips=UNSET, org_id=UNSET, current_datetime=No
 
     @property
     def is_active(self):
+        # if accessing differnt org. Also can't use self.org_id as ORG_ID is None
```
what do you mean by this comment? ORG_ID is None?

maxiuyuan committed:
compare with self.org_id

maxiuyuan committed:
fixed tests

maxiuyuan committed:
redo last commit

maxiuyuan committed:
fixed tests

maxiuyuan committed:
Merge branch 'log_su_access' into su_expiry

k-fish committed:
fix(perf): Add try/catch around observer (#32579)

This is the second time we've had an error on Safari around the perf code, going to wrap in a try/catch since the code is auxiliary and shouldn't throw errors.

edwardgou-sentry committed:
fix(discover): Add to Dashboard test from a Discover Query (#32577)

Add to Dashboard tests from a Discover Query using RTL and jest.mock

dashed committed:
fix(design): Fix code css definition (#32589)

edwardgou-sentry committed:
fix(dashboards): Don't display edit button in Widget Viewer for prebuilt dashboards (#32567)

Disables displaying the edit button for Widget Viewer for prebuilt dashboard widgets

wedamija committed:
fix(tests): Another attempt to reduce flakiness of event frequency tests (#32580)

Fixing the date at a specific time of day and also making sure that freeze_time is used overall for
the test and not at multiple points.

Also improved how we create sessions

vuluongj20 committed:
ref(page-filters): Add `alignDropdown` prop to environment selector (#32461)

* ref(page-filters): Add alignDropdown prop to environment selector

* ref(page-filters): Add alignDropdown prop to environment selector

* style(lint): Auto commit lint changes

Co-authored-by: getsantry[bot] <66042841+getsantry[bot]@users.noreply.github.com>

vuluongj20 committed:
ref(ui): Make small UI changes to Button component (#31903)

* ref(ui): Make small UI changes to Button component

* fix(ui): Fix wrong parentheses in logical operators in <DropdownButton />

* ref(ui): Use darker hover background for buttons

* fix(settings): Fix panel overflow in integration project settings

robinrendle committed:
fix(ui): copy of alert details chart (#32587)

robinrendle committed:
fix(ui): fixed size of open in issues button (#32573)

vuluongj20 committed:
fix(projects): Use subText for project card's score title (#32595)

* fix(projects): Use subText for project card's score title

* style(lint): Auto commit lint changes

Co-authored-by: getsantry[bot] <66042841+getsantry[bot]@users.noreply.github.com>
Co-authored-by: Robin Rendle <robinjrendle@gmail.com>

vuluongj20 committed:
fix(projects): Use subText for project issues table's heading (#32597)

Commit:
feat(codeowners): Decode base64 CODEOWNERS contents to utf-8 (#32582)

Fixes SENTRY-TH4

Commit:
feat(onboarding): Add onboarding welcome page experiment (#31733)

* adds onboarding welcome page experiment

* temp checkin

* adds tests

* adds onboarding images

* making progress

* adds falling man

* Use browser history

* Hanging man position & viewport height

* Add call to logExperiment

* Falling man text adjustments

* Update static/app/views/onboarding/onboardingController.tsx

Co-authored-by: Evan Purkhiser <evanpurkhiser@gmail.com>

* fixes

* Disable logExperimetn

* fix tests

Co-authored-by: Zhixing Zhang <me@neoto.xin>
Co-authored-by: Evan Purkhiser <evanpurkhiser@gmail.com>

evanpurkhiser committed:
chore(js): Upgrade babel from 7.15 -> 7.17 (#32016)

dependabot[bot] committed:
chore(deps): bump pillow from 8.3.2 to 9.0.1 (#32552)

evanpurkhiser committed:
test(js): Convert organizationGroupDetails/action to RTL (#32598)

evanpurkhiser committed:
ref(ui): Improve uage of BookmarkStar (#32553)

sl0thentr0py committed:
chore: Add team-web-backend as CODEOWNERS of SDK (#32033)

edwardgou-sentry committed:
feat(dashboards): Widget Viewer beta badge and style (#32588)

Beta Feature Badge and slight improvement to Multi Query text style

k-fish committed:
ref(sdk): Move back to 5000ms idleTimeout (#32564)

This removes the push to a higher idleTimeout as well as disabling the reportAllChanges flag to ensure 5000ms is a decent default.

wmak committed:
feat(mep): Introduce the count_web_vitals functions to metrics (#32507)

- These are the equivalent metrics function versions of what was added
  in discover

wmak committed:
feat(mep): Introduce p100 to the metric query builder (#32570)

- This introduces "p100" which is the same as doing a max to the metric
  query builder

shruthilayaj committed:
ref(dam): Use cols and aggregates (#32441)

Read from columns and aggregates instead of fields to
build widgets. Fields are still used to maintain order on
table widgets.

JoshFerge committed:
feat(devenv): set sentry SDK default off if dsn not provided and relay off (#32569)

Co-authored-by: Alberto Leal <mail4alberto@gmail.com>

k-fish committed:
feat(perf): Add MEP param to other landing requests (#32575)

* feat(perf): Add MEP param to other landing requests

This will add the metricsEnhanced param to the other widget and table requests, since it should be safe to add even if it's not consumed.

getsentry-bot committed:
release: 22.3.0

Zylphrex committed:
ref(snuba): Use keyword args for snuba column names (#32523)

Snuba columns were being declared using a named tuple using position arguments
to initialize. This meant you had to know the order of the named tuple to map to
which dataset the name is for. This uses the keyword arguments supported by
named tuples to be explicit which dataset the name is for.

volokluev committed:
Shrink rate limit (#32585)

It's all part of the plan baby

0Calories committed:
Add feature flag for autogrouping sibling spans (#32571)

edwardgou-sentry committed:
fix(dashboards): Optional fieldValue for queryField (#32623)

checks fieldValue on render which can sometimes be undefined due to ui.

scttcper committed:
feat(workflow): Track alert rule edited events (#32600)

JonasBa committed:
ref(test): throw if a fixture ever gets overriden (#32551)

* ref(test): throw if a fixture ever gets overriden

* ref(test): drop comments

* ref(test): use the right fixture type

Commit:
feat(ui-components): Rm feature flag for Alert Rule Actions (#32601)

Zylphrex committed:
Revert "test(js): Convert organizationGroupDetails/action to RTL (#32598)" (#32633)

This reverts commit 170de5484886e598efe2736c281c1f4a7e3b3816.

wmak committed:
feat(mep): Adds misery & apdex functions (#32599)

* feat(mep): Adds misery & apdex functions
- This introduces the misery, count_miserable and apdex functions
- Gonna follow up with more tests soon

wmak committed:
feat(mep): Add the team-key-transaction alias (#32593)

- This moves some of our aliases to their own files, the intention is to
  eventually migrate more of the class methods there this gives us a few
  benefits
  - We can unit test the resolverse now
  - They're easier to re-use across datasets

Zylphrex committed:
fix(profiling): Move flame graph tooltip above cursor (#32495)

When the cursor is on the left half of the screen, the tooltip appears below to
the right of the cursor. However, the cursor also covers the first few
characters in the tooltip. This change shifts the tooltip down a little so that
the text is not covered.

billyvg committed:
feat(ci): *TEMP* downgrade GHA runners for acceptance tests to 18.04 (#32641)

See https://github.com/dotnet/core/issues/7286

dcramer committed:
feat: Pass relay status via environment variable (#32629)

scttcper committed:
fix(alerts): Fix loading for orgs with many projects (#32627)

edwardgou-sentry committed:
feat(dashboards): Widget Viewer updates table columns Title and the yAxis value (#32568)

Changes the Table columns for Line, Area, Bar, and Big number widgets in the Widget Viewer.
Just displays Title and the Y-Axis value now, since the previous columns we displayed were not always universally applicable

edwardgou-sentry committed:
feat(dashboards): Widget Viewer Analytics (#32594)

Adds analytics to widget viewer events

dependabot[bot] committed:
chore(deps): bump url-parse from 1.4.7 to 1.5.10 (#32106)

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

evanpurkhiser committed:
ref(js): Avoid usage of deprecated ReactNodeArray type (#32610)

evanpurkhiser committed:
chore(deps): Bump stylelint (#32615)

evanpurkhiser committed:
ref(js): Extract CellAction makeCellActions (#32612)

evanpurkhiser committed:
chore(deps): Bump color from 3 -> 4 (#32614)

k-fish committed:
fix(perf): Check MEP response is not undefined (#32631)

This will include 'undefined' (such as before the metric data is returned) to not show sampled. Sampled will only then be shown if isMetricsData is explicitly true.

PixelDust22 committed:
fix(onboarding): Onboarding welcome screen css fixes (#32643)

* Falling Error

* update styles

* more updates

dashed committed:
fix(design): Separate background colours for code and pre (#32638)

evanpurkhiser committed:
chore(deps): Bump dompurify (#32613)

evanpurkhiser committed:
deps(js): Bump prettier 2.3 -> 2.5 (#32608)

Commit:
Merge branch 'releases/22.3.0'

Commit:
meta: Bump new development version

evanpurkhiser committed:
style(js): Minor cleanup in utils/dates (#32616)

dependabot[bot] committed:
build(deps): bump follow-redirects from 1.5.10 to 1.14.8 (#31780)

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps): bump prop-types from 15.7.2 to 15.8.1 (#32032)

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps): bump webpack-cli from 4.7.2 to 4.9.2 (#31991)

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps-dev): bump html-webpack-plugin from 5.3.2 to 5.5.0 (#30871)

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

scttcper committed:
fix(alerts): Report correct number of issues in digest email (#32550)

evanpurkhiser committed:
style(js): Reduce indentation in tagsHeatMap (#32607)

ahmedetefy committed:
ref(metrics): Clean up tests (#32637)

* ref(metrics): Clean up tests

* Trigger Build

* Added freeze_time to flaky tests

* Adds missing super setup call

* Fix failing test due to freeze_time

dependabot[bot] committed:
chore(deps): bump papaparse from 5.3.1 to 5.3.2 (#32644)

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

evanpurkhiser committed:
ref(ui): Try and fix flakey visual regression on breadcrumbs (#32646)

ahmedetefy committed:
ref(metrics): Refactor tests into their own modules (#32640)

dependabot[bot] committed:
chore(deps): bump core-js from 3.16.4 to 3.21.1 (#32653)

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps-dev): bump storybook-dark-mode from 1.0.8 to 1.0.9 (#32645)

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps): bump less-loader from 10.0.1 to 10.2.0 (#32652)

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

Commit:
fix(page-filters): Fix double selection issue (#32657)

dependabot[bot] committed:
chore(deps): bump pyright from 1.1.186 to 1.1.229 (#32659)

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps-dev): bump typescript-styled-plugin from 0.18.1 to 0.18.2 (#32664)

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

mikellykels committed:
fix(ui): Show timezone with 24hr clock event info on issue details (#32611)

When a user has the 24 hour clock set, the date/time of the event on the issue details page didn't display the timezone. This adds the timezone.

FIXES WOR-739

evanpurkhiser committed:
chore(deps): bump fork-ts-checker-webpack-plugin from 6.3.3 to 7.2.1 (#32672)

armenzg committed:
feat(ci): Make dev env check faster by splitting off pyenv testing (#32565)

Separate `pyenv` set up logic into its own check to speed up the dev env set up.
There's some other caching fixes that will follow up on another PR.

Without this change, it can take close to 30 minutes:
<img width="248" alt="image" src="https://user-images.githubusercontent.com/44410/158403860-1a3a7b18-0bd9-4c34-af6e-afb83a4e008e.png">

With this change, it can run closer to 20 minutes:
<img width="251" alt="image" src="https://user-images.githubusercontent.com/44410/158403778-94bc47a7-847f-402f-9344-a025a54c8130.png">

evanpurkhiser committed:
fix(ui): Share link as FC + typography (#32666)

evanpurkhiser committed:
ref(js): Improve NotAvailable component (#32673)

billyvg committed:
feat: Upgrade `jest-sentry-environment`@1.4.0 (#32656)

This fixes certain transactions missing branch/ref tags and changes transaction status to be `internal_error` when tests fail.

See https://github.com/getsentry/jest-sentry-environment/pull/1

wmak committed:
ref(discover): move field and filter aliases to datasets (#32660)

- This also moves some of the functions from datasets base to the alias
  files instead

narsaynorath committed:
test(acceptance): Fix flaky big number in dashboard tests (#32688)

It isn't important that a big number is the second widget, so we can
swap it out for a different chart that's rendered with getDynamicText

evanpurkhiser committed:
chore(deps): Bump @types/* only packages (#32679)

wmak committed:
fix(mep): Update meta when reverse resolve tag values (#32655)

- This updates meta so when we change tags from ints back to their
  strings we also update the meta from int to string

evanpurkhiser committed:
fix(js): Mark FormModel.resetForm as a mobx action (#32682)

evanpurkhiser committed:
chore(deps): Upgrade echarts 5.3.0 -> 5.3.1 (#32684)

evanpurkhiser committed:
chore(deps): Remove unused csstype (#32678)

evanpurkhiser committed:
chore(deps): Bump mini-css-extract-plugin 2.0.0 -> 2.6.0 (#32683)

evanpurkhiser committed:
chore(deps): Bump react-grid-layout from 1.3.0 -> 1.3.4 (#32677)

Commit:
ref(gsh): Change projects prop to memberProjects (#32658)

changed to memberProjects to be more descriptive of what the prop actually represents.

udameli committed:
feat(perf): Add base spans histogram endpoint (#32529)

* feat(perf): Add events spans histogram url

* feat(spans): Add spans histogram endpoint

* fix typos

* Add spans histogram query

* clean up the spans histogram query

* clean up test data

* make endpoint private

* add base tests

* udpate tests

* exclude spans histogram endpoint from the public list

* make endpoint private

* update codeowners

* use spans.exclusive_time in SnQL Function for spans histograms

* clear selected_columns

* document spans histogram query params

* handle quotes and backslashes in span ops

* rename args in span histogram snql function

* add proper tests

* add spans histogram function to the query builder

* more tests

evanpurkhiser committed:
chore(deps): Upgrade webpack-remove-empty-scripts 0.7.1 -> 0.7.3 (#32681)

evanpurkhiser committed:
chore(deps): Bump prettier 2.5.1 -> 2.6.0 (#32680)

edwardgou-sentry committed:
feat(dashboards): Widget Viewer persistent column widths (#32624)

Makes table column widths in the Widget Viewer persistent between link sharing and Widget Viewer interactions.
Column widths are stored in the url as query parameters.
This also solves existing issues with column widths being reset on pagination, query selection, legend toggling, etc

robinrendle committed:
fix(ui): spacing problems on the issue details page (#32667)

* fix(ui): spacing problems on the issue details page

* more spacing fixes

* Update static/app/components/group/releaseStats.tsx

Co-authored-by: Evan Purkhiser <evanpurkhiser@gmail.com>

* GraphSpacing -> GraphContainer

Co-authored-by: Evan Purkhiser <evanpurkhiser@gmail.com>

scttcper committed:
fix(ui): Remove route re-mount when orgId is undefined (#32668)

edwardgou-sentry committed:
fix(dashboards): Widget Viewer better snapshot test (#32693)

Makes the query selector in snapshot tests not as ugly

edwardgou-sentry committed:
chore(discover): Moves saved query tests from jsx to tsx (#32692)

Moves the remaining jsx tests from saved query to tsx.

Commit:
feat(vercel): allow a single Sentry project to map to multiple Vercel projects (#32639)

This PR allows a single Sentry project to map to multiple Vercel projects as some users have requested this. There was no real reason we prevented this before. Note that it's not possible to support multiple Sentry projects for a single Vercel project.

I also swapped the columns so the Vercel project is on the left which makes it less confusing when you are mapping to the same Sentry project:

dashed committed:
fix(span): Fix end timestamp of nested grouped spans (#32702)

Commit:
fix(page-filters): Fix potential overflow on user feedback (#32705)

PixelDust22 committed:
chore(reports): Turn on weekly email tasks instrumentation (#32699)

Commit:
fix arrow location (#32710)

scttcper committed:
feat(workflow): Remove guides "for review", assigned suggestion (#32503)

scttcper committed:
ref(ui): Convert issue stream header to RTL, useProjects (#32510)

evanpurkhiser committed:
fix(page-filters): Correct bug in environment quick-select (#32712)

mgaeta committed:
ref(endpoints): Pass OrganizationMember through `convert_args()` (#32718)

priscilawebdev committed:
ref(new-widget-builder-experience): Move widget builder step components to separate files (#32559)

evanpurkhiser committed:
ref(ui): Convert settings ProjectFilters to a FC (#32720)

priscilawebdev committed:
ref(test): Update flakey custom repository test (#32725)

beezz committed:
fix(metrics): Fix misplaced cache_hit metric tag assignment (#32730)

ahmedetefy committed:
fix(metrics): fix bug in tags and tags values endpoint (#32590)

Fixes bug where a single
incorrect setup derived metrics would not be caught
in tag and tag details because we were exiting the
loop once we find a match in one of the entities
regardless of whether there was another metric
constituent in another entity

mitsuhiko committed:
meta: Issue template for flaky tests (#32726)

armenzg committed:
feat(dev): Error reporting for pre-commit and improvements (#32522)

When changes for `.envrc` happen, we need to execute `direnv allow`, otherwise, we don't activate the Python 3 venv.

When we `git pull` and there are `.envrc` changes then direnv deactivates. This means that Python 2 (the system's default) becomes active, thus, Sentry is not installed inside of a venv.

We will now let the developers know when executing post-merge hooks:
* that they should execute `direnv allow` if not using Python 3.
* that they should execute `make install-py-dev` if Sentry is not installed in the venv.
* that errors will not be reported unless they have `SENTRY_DEV_ENV_DSN` set

Co-authored-by: Evan Purkhiser <evanpurkhiser@gmail.com>

shruthilayaj committed:
fix(dashboards): Return an empty array if null (#32691)

Returns an empty array if value is null.

Fixes JAVASCRIPT-26TN

priscilawebdev committed:
ref(new-widget-builder-experience): Split components feedback (#32729)

mgaeta committed:
ref(rules): Explicit Rule IDs (#32516)

ahmedetefy committed:
feat(metrics): Return crash_free_rate as a ratio (#32734)

Modifies the SnQL for crash_free_rate to return
the value as a ratio rather than a percentage

robinrendle committed:
fix(ui): remove new tag (#32701)

edwardgou-sentry committed:
feat(discover): Track discover query conditions (#32700)

Track analytics for Discover queries submitted through the Discover results page

volokluev committed:
ref(ratelimits): surface concurrent requests to access logs (#32301)

### Context
All context on this project and reasoning for the implementation can be found [here](https://www.notion.so/sentry/Concurrent-Rate-Limiter-ebd5b7562aed4d01b102d83774a06b11)

This is the first step of concurrent rate limits in sentry. In order to understand what to rate limit by, we fist need to understand what the concurrent usage of our API is. 

### This PR does the following:

* Implements the concurrent rate limiting mechanism which will be used in sentry
* Surfaces the data to the access logs
* Does NOT enforce any concurrent rate limits
* Surfaces concurrent limit data in headers

Some unrelated but otherwise important things:
* Makes the rate limit middleware fail open. As it is, an exception in that middleware would shut down the request

PixelDust22 committed:
fix(onboarding): Onboarding illustration style changes (#32741)

* Onboarding styles

* css adjustmentes

mgaeta committed:
ref(rules): Fix abstract classes (#32500)

mgaeta committed:
ref(tests): DRY OrganizationMember tests. (#32715)

k-fish committed:
fix(perf): Switch MEP sampled condition  (#32698)

* fix(perf): Switch MEP sampled condition

Switch sampled tag so it appears on isMetricsData false.

vuluongj20 committed:
ref(page-filters): Use IconLock for pin feature (#32635)

* ref(page-filters): Use IconLock for pin feature

* ref(page-filters): Remove pin button tooltip delay

* fix(page-filters): Fix tests

* ref(page-filters): Make pin button icon smaller

* fix(page-filters): Fix tests

* ref(page-filters): Remove tooltip around pin button

dashed committed:
chore(perf): Rename variable names in span view files (#32663)

volokluev committed:
make rate limit middleware fail open on process_response (#32743)

* make rate limit middleware fail open on process_response

* test coverage

Co-authored-by: Vlad Kluev <vlad.kluev@sentry.io>

narsaynorath committed:
feat(new-widget-builder-experience): Group by field for single y-axis query (#32649)

JonasBa committed:
ref(profiling): move flamegraph search state (#32581)

* feat(flamegraph): add state provider

* feat(flamegraph): use state provider and hook

* feat(flamegraph): wire up navigation

dashed committed:
fix(discover): Fix flakey test on span view in Discover (#32733)

Co-authored-by: Ash Anand <0Calories@users.noreply.github.com>

leeandher committed:
chore(api): Remove update teams code from project details (#32632)

See getsentry/sentry-docs#4813

This was already mostly done but the messaging that it's deprecated has lived on the endpoint for around four years now. I've opted to just remove the deprecated code all together, but just the docs change is important. I can remove the actual changes to the endpoint if we want to preserve that messaging.

Zylphrex committed:
fix(profiling): Allow switching threads on imported profile (#32650)

The thread selector index only operated on the profile from the props.

vuluongj20 committed:
ref(page-filters): Use DropdownAutocomplete for TimeRangeSelector (#32349)

* ref(page-filters): Use DropdownAutocomplete for TimeRangeSelector

* fix(test): Fix tests related to TimeRangeSelector

* fix(page-filters): Remove hidePin props

* ref(time-range-selector): Use new SelectorItemsHook

* fix(page-filters): Add allowActorToggle prop to TimeRangeSelector

* ref(page-filters): Remove SelectorItems file

* fix(time-range-selector): Fix appearance in Stats & Releases

* fix(dropdown-autocomplete): Fix bubble's min-width

* fix(dropdown-autocomplete): Add min-width to content

* ref(page-filters): Use inset box-shadow for filter bar border

JoshFerge commented on src/sentry/auth/superuser.py:
```diff
@@ -43,6 +43,11 @@
 
+# the maximum time the session can stay alive when accessing different orgs
+MAX_AGE_PRIVILEGED_ORG_ACCESS = getattr(
+    settings, "SUPERUSER_ORG_CHANGE_MAX_AGE", timedelta(minutes=15)
```
should this setting string be `MAX_AGE_PRIVILEGED_ORG_ACCESS`?

JoshFerge commented on src/sentry/auth/superuser.py:
```diff
@@ -70,6 +75,15 @@ class Superuser:
     org_id = ORG_ID
 
+    def _check_expired_on_org_change(self):
+        if hasattr(self, "expires") and self.expires is not None:
```
why do you need to check if this attribute exists with `hasattr`?

maxiuyuan replied:
i  added this since expires is not initialized at the beginning and only set when _set_logged_in is called. I added this since certain test cases before tested the expiry time without actually calling set_logged_in, in which this would error out since self.expires doesn't exist. As far as i know, i don't think it would happen on prod but as it happens in the tests i was paranoid

JoshFerge commented on tests/sentry/auth/test_superuser.py:
```diff
@@ -170,6 +170,21 @@ def test_su_access_logs(self, logger):
+
+        assert superuser.is_active is True
+
+    def test_max_time_org_change_time_expired(self):
```
doing something like `freeze_time` instead of setting the expiry on the su object itself is more realistic to the flow of a user, so converting the test to do that is preferable.

JoshFerge commented on src/sentry/auth/superuser.py:
```diff
@@ -82,6 +96,9 @@ def __init__(self, request, allowed_ips=UNSET, org_id=UNSET, current_datetime=No
     @property
     def is_active(self):
+        org = getattr(self.request, "organization", None)
+        if org and org.id != self.org_id:
```
when would the org not exist? and can we add a test case for that if so?

maxiuyuan replied:
I've noticed that this will be hit on everything that relies on the superuser class. since is_active_super() is called quite a bunch. if i put a breakpoint here, its hit even on login, during that time there isn't an org in the request, so thats when it doesn't exist. ill add a test case for this

mgaeta committed:
ref(types): Add types to rules (#31590)

Co-authored-by: getsantry[bot] <66042841+getsantry[bot]@users.noreply.github.com>

evanpurkhiser committed:
ref(ui): Convert Tooltip to a FC (#32557)

scttcper committed:
feat(workflow): Add skipInitializeUrlParams to global selection (#32716)

Allows forced projects to not be persisted in url parameters on load
fixes https://getsentry.atlassian.net/browse/WOR-1691

mikellykels committed:
fix(workflow): Handle zoom on metric alert details chart (#32723)

This fixes an issue where zoom wasn't working on the metric alert details chart. With onFinished on the AreaChart, this blocked ChartZoom from being access/updated.

FIXES WOR-1596

mgaeta committed:
fix(getsentry): `RuleFuture` circular import (#32745)

ahmedetefy committed:
chore: Update CODEOWNERS for metrics team (#32749)

jjbayer committed:
fix(metrics): Different treatment for unresolved strings and absent tags (#32574)

We currently map strings that are missing from the indexer to an integer
index of 0 in resolve_weak, so we don't have to deal with missing
strings on Sentry side. This works well when all stored buckets have a
tag value for a given tag key, that is, WHERE tags[123] = 0 will return
an empty result.

But when there are buckets which do not have a value for tags[123] at
all, the above condition will resolve to those untagged buckets rather
than an empty result set.

This PR resolves that ambiguity by inserting -1 rather than 0 into the
query.

ahmedetefy committed:
fix(metrics): Fix 500 error with specific interval and statsPeriod combination (#32742)

Fixes a bug that was throwing a 500 error
for specific statsPeriod and interval combinations
because the paginator sets a default limit
of 51, and then if the interval is too
low compared to the statsPeriod, we might be
exceeding the snuba 10,000 query limit
as an example, if we request statsPeriod of 24h
and interval of 5 min, we end up with 288 intervals
for time series query, which when we multiply by 51
we end up with 14,688

Commit:
ref(js): Change rules/filter to FC (#32761)

changed to FC

matejminar committed:
ref(metrics): Cleanup metrics to discover helpers (#32483)

matejminar committed:
fix(ui): Fix timerange autocomplete (#32767)

jjbayer committed:
feat(release_health): Capture function args in duplex (#32768)

Make sure we always capture all function args for release health
functions when reporting differences in the duplex implementation.

shruthilayaj committed:
chore(dam): Rename Release Health Data Set selector (#32621)

Rename Metrics dataset selector to Release Health

k-fish committed:
fix(tests): Replace eCharts in landing test (#32732)

Replaces the visualizations with static placeholder until we fix the ingestion store time issue with echarts

JonasBa committed:
feat(utils): add undoable and combined reducer (#32622)

* feat(reducers): add useCombinedReducer

* feat(reducers): add undoableReducer

* feat(profiling): wire up state and undo/redo functionality (#32634)

* feat(reducers): wire the state

* feat(reducers): use the right import

* ref(flamegraph): dispatch payload

* ref(search): remove React.* and explicitly check for undefined state

* ref(search): fix bad rebase

narsaynorath committed:
feat(new-widget-builder-experience): Multiple y-axes with grouping (#32721)

udameli committed:
ref(spans): Normalize exclusive time histogram results (#32762)

* ref(spans): Normalize exclusive time histogram results

* test normalized data

ahmedetefy committed:
feat(metrics): Support for DM in Details Endpoint (#32744)

Adds support for derived metrics in metrics detail
 endpoint

edwardgou-sentry committed:
feat(dashboards): Widget Viewer multi query style (#32735)

Updates multi query text and fixes dropdown

wmak committed:
fix(mep): Allow sorting by alias (#32706)

- This fixes a sorting bug on performance, where we use the alias of
  functions to sort as well as the full name
- Wasn't working cause we'd try to resolve the column of a function
  alias like `tpm`, but this would raise an Incompatible Metrics Query
  error

wmak committed:
fix(mep): Add title as a field to MEP (#32709)

- This allows title to be used via MEP as well, since its only
  transaction data we can just alias it to the transaction field
- This also fixes the typing on SelectType to include AliasedExpression

wmak committed:
fix(mep): Handle invalid metric in parameters (#32713)

- This handles the case where the parameter to a function isn't valid in
  metrics, eg. count_unique on a tag
  - This was raising an Invalidsearch error which meant we weren't
    trying the query again in Discover

leeandher committed:
fix(ui): Adjust formatting on webhook checkboxes (#32750)

See ISSUE-1429

This PR just adjusts how the CSS is applied to the webhook checkboxes on the create integration pages. It also changes the tootltip to be a bit more accurate (e.g. there is no 'Comment' permission to enable, we're looking for 'Event').

wmak committed:
feat(mep): Add an allowlist for each metric function (#32748)

- This is so that we don't allow something like p75(user), since `user`
  will resolve to a valid metric id

MeredithAnya committed:
ref(metrics-indexer): Add new indexer table (#32540)

* ref(metrics-indexer): Add new indexer table

narsaynorath committed:
fix(new-widget-builder-experience): Issues widgets use issues search bar (#32754)

scttcper committed:
test(ui): Convert saved search menu to RTL (#32513)

mikellykels committed:
fix(workflow): Cancel request on query change for issue list tab counts (#32675)

* fix(workflow): Cancel request on query change for issue list tab counts

Previously, it could be possible to show the incorrect tab count when changing projects or query. This PR will cancel the request for fetchCounts.

FIXES WOR-663

* reorder function to read easier

mikellykels committed:
feat(workflow): Add assign to Me to top of assignee dropdown (#32602)

* feat(workflow): Add assign to Me to top of assignee dropdown

Add Me to the top of the assignee dropdown.

FIXES WOR-524

* fix for tests, move Clear Assignee down to footer and update design

* move session user to top of suggested and update to show name with (You)

robinrendle committed:
fix(ui): issue tag layout fixes (#32760)

* fix(ui): issue tag layout fixes

* removed useless imports

* fixed test

* change test name

PixelDust22 committed:
fix(onboarding): Add onboarding sandbox btn cta & logExperiment (#32759)

* fix(onboarding): Add  onboarding sandbox btn cta

* Turn on experiment assignment

phacops committed:
feat(profiling): Pull from Kafka topic to run symbolication (#32320)

maxiuyuan committed:
Merge branch 'log_su_access' into su_expiry

maxiuyuan committed:
freezetime for tests and changed setting variable

billyvg committed:
feat(replays): Allow rrweb player to stitch together multiple/partial replays (#32536)

Changes rrweb player to fetch all replays that start with `rrweb` and merge
events together before passings events to the player.

We're currently prototyping something that kind of emulates streaming of rrweb events. Currently, we assume that `rrweb.json` has all of the events that are needed for the player to replay. We want to change that a bit so that instead of one large JSON file of events, they are broken up into multiple JSON files with events. (Another assumption we're making is that they sorted chronologically, which should be the case as we append a timestamp to each JSON file).

AniketDas-Tekky committed:
fix(stats): Don't count token requests in FE stats (#32752)

* fix(stats): Don't count token requests in FE stats

When looking at the frontend stats that we had been getting, some of the cookie requests also had bearer tokens too. These are most likely not done by the UI.
I added this check to reduce some noise on the "UI" request counts.

* Added docstrings

maxiuyuan committed:
added test to see if org is empty

Commit:
fix(page-filters): Show pin only with new page filters (#32462)

Realized that I feature flagged the pin button, which meant it showed on global selection header dropdowns even when they weren't new page filters. Added a prop which is sent from page filter components to the dropdowns.

Commit:
ref(page-filters): Improve checking for new paths with page filters (#32509)

Use includes for the pathname to avoid having to hardcode each subpage ex: alerts/rules and alerts/

dcramer committed:
feat: Improve relay debug in CI (#32625)

Commit:
fixed (#32789)

iker-barriocanal committed:
fix(metrics): Zerofill only on groupby session.status (#32775)

Zero-filling when there isn't any data must only happen when there's a
single groupBy property, and that property is session.status. This PR
fixes a bug where the data is zero-filled when session.status is
present, even if it isn't the only property. The test has also been
updated to consider every combination of session.status values.

matejminar committed:
feat(ui): Clamp release period to 90 days  (#32769)

jjbayer committed:
test(anomaly-detection): Freeze time (#32799)

Backend tests started failing because the 90 day limit had passed for
the given test case.

narsaynorath committed:
feat(new-widget-builder-experience): Stack bar chart when grouping (#32770)

jjbayer committed:
fix(release-health): Don't compare get_project_sessions_count (#32792)

We verified the correctness of the metrics implementation of
get_project_sessions_count manually. The results still differ because
the sessions implementation gets its results from hourly aggregations.

0Calories committed:
feat(span-tree): Autogroup sibling spans (#32221)

When several consecutive sibling spans in the Span Tree share the same operation in description, it can make it tedious to scroll through the tree, and these repeat spans don't provide much value. This PR allows these spans to be autogrouped, with the option to expand/collapse them.

Addresses VIS-1427

JonasBa committed:
ref(tsc): remove HTMLProps type intersection (#32697)

* ref(tsc): remove HTMLProps type intersection

* ref(hovercard): make handlers optional

* ref(props): alias props and use HTMLAttributes

* fix(tsc): use correct htmlattributes

scttcper committed:
fix(issues): Set current release stats correctly (#32784)

ahmedetefy committed:
ref(metrics): Refactor repeated code into its own func (#32791)

* ref(metrics): Refactor repeated code into its own func

Refactors similar logic in get_tags, get_tag_values,
and get_single_metric_info into its own aux function
_fetch_tags_or_values_per_ids

* PR feedback

JoshFerge commented on tests/sentry/auth/test_superuser.py:
```diff
@@ -189,6 +193,30 @@ def test_su_access_no_request(self, logger):
                 "superuser.logged-in", extra={"ip_address": "127.0.0.1", "user_id": 10}
             )
 
+    @freeze_time("2022-03-03 08:21:34")
```
could you reconstruct the arguments to freeze_time as the base time + a timedelta? it is hard for me to follow otherwise.

JoshFerge commented on tests/sentry/auth/test_superuser.py:
```diff
@@ -189,6 +193,30 @@ def test_su_access_no_request(self, logger):
             )
 
+    @freeze_time("2022-03-03 08:21:34")
+    def test_expired_check_org_in_request(self):
+        request = self.build_request(expires=self.current_datetime)
+        superuser = Superuser(request, allowed_ips=())
+        assert superuser.is_active is False
+        assert not getattr(request, "organization", None)
```
i feel like a test where superuser would be active otherwise would be a better test for this behavior (outside of org change limit but otherwise active with no org in the request)

maxiuyuan committed:
changed freeze time with basetime and delta and changed test behavior, removed checking expired as attribute

maxiuyuan committed:
Merge branch 'master' into su_expiry
